From 10431604666baa5b5c3e71084e76f6337f9ec3ac Mon Sep 17 00:00:00 2001
From: Mingcong Bai <jeffbai@aosc.io>
Date: Mon, 7 Apr 2025 22:29:50 +0800
Subject: [PATCH 04/14] AOSCOS: media: enable VA-API decoding/encoding by
 default

Enable the following features by default:

- AcceleratedVideoEncoder
- AcceleratedVideoDecoder
- AcceleratedVideoDecodeLinuxGL
- VaapiOnNvidiaGPUs

Link: https://chromium.googlesource.com/chromium/src/+/refs/heads/main/docs/gpu/vaapi.md
Signed-off-by: Mingcong Bai <jeffbai@aosc.io>
---
 media/base/media_switches.cc | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/media/base/media_switches.cc b/media/base/media_switches.cc
index f23a85d6ad..3a22a7b159 100644
--- a/media/base/media_switches.cc
+++ b/media/base/media_switches.cc
@@ -680,19 +680,15 @@ BASE_FEATURE(kUnifiedAutoplay,
 // on chromeos, but needs an experiment on linux.
 BASE_FEATURE(kAcceleratedVideoDecodeLinux,
              "AcceleratedVideoDecoder",
-#if BUILDFLAG(USE_VAAPI)
              base::FEATURE_ENABLED_BY_DEFAULT);
-#else
-             base::FEATURE_DISABLED_BY_DEFAULT);
-#endif
 
 BASE_FEATURE(kAcceleratedVideoDecodeLinuxGL,
              "AcceleratedVideoDecodeLinuxGL",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 BASE_FEATURE(kAcceleratedVideoEncodeLinux,
              "AcceleratedVideoEncoder",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Ignore the non-intel driver blacklist for VaapiVideoDecoder implementations.
 // Intended for manual usage only in order to gague the status of newer driver
@@ -706,7 +702,7 @@ BASE_FEATURE(kVaapiIgnoreDriverChecks,
 // crashes, disable VA-API on NVIDIA GPUs by default. See crbug.com/1492880.
 BASE_FEATURE(kVaapiOnNvidiaGPUs,
              "VaapiOnNvidiaGPUs",
-             base::FEATURE_DISABLED_BY_DEFAULT);
+             base::FEATURE_ENABLED_BY_DEFAULT);
 
 // Enable VA-API hardware low power encoder for all codecs on intel Gen9x gpu.
 BASE_FEATURE(kVaapiLowPowerEncoderGen9x,
-- 
2.49.0

